cmd/libsnap-confine-private/mount-opt-test.c

```diff
@@ -224,8 +227,15 @@ static void test_sc_do_mount(void)
 	}
 	g_test_trap_subprocess(NULL, 0, 0);
 	g_test_trap_assert_failed();
-	g_test_trap_assert_stderr
-	    ("cannot perform operation: mount -t ext4 -o ro /foo /bar: Permission denied\n");
+	if (GPOINTER_TO_INT(snap_debug) == 0) {
+		g_test_trap_assert_stderr
+		    ("cannot perform operation: mount -t ext4 -o ro /foo /bar: Permission denied\n");
+	} else {
+		/* make sure we don't end up hiding the error if debug logging is enabled */
```

Comment:
Thanks for adding those tests! 
I would expand this comment a little, I was struggling with this because I found it confusing that things get hidden and then things don't get hidden. After careful reading I now see that one if the debug output and one is the error output. So maybe something like:
```
                /* - without snap_debug the debug output hides the actual mount commands
		 * - *but* the actual mount error is displayed
		 */
```